src/main/java/sirius/biz/sequences/Sequences.java

```diff
@@ -0,0 +1,176 @@
+@Framework("sequences")
+public class Sequences {
+
+    public static final Log LOG = Log.get("sequences");
+
+    @Part
+    private OMA oma;
+
+    /**
+     * Returns the next value in the given sequence.
+     * <p>
+     * Note that this method doesn't use locks or transactions. It rather utilizes optimistic locking, which scales
+     * extremely well. However, the algorithm used is not intended for extreme parallel usage. In such scenarios,
+     * it will not content but rather give up after some tries and report an appropriate exception.
+     *
+     * @param sequence the name of the sequence which is counted up.
+     * @return the next value (which has not yet been returned). If the sequence is unknown, <tt>1</tt> is returned.
+     * @throws HandledException If the system was unable to generate a unique sequence number. This might happen in
+     *                          extreme load conditions, as internally an optimistic locking algorithm is employed.
+     */
+    public long generateId(String sequence) {
+        try {
+            int retries = 25;
+            while (retries-- > 0) {
+                // Select the current value which will be returned if all goes well....
+                SequenceCounter result =
+                        oma.select(SequenceCounter.class).eq(SequenceCounter.NAME, sequence).queryFirst();
+                if (result == null) {
+                    try {
+                        // Try to create a new record, as no counter is yet present...
+                        result = new SequenceCounter();
+                        result.setName(sequence);
+                        result.setNextValue(2);
+                        oma.update(result);
+                        return 1;
+                    } catch (HandledException e) {
+                        // This only happens if another thread / server inserted the entity already...
+                        Exceptions.ignore(e);
+                        continue;
+                    }
+                }
+
+                int numRowsChanged = oma.getDatabase()
+                                        .createQuery("UPDATE sequencecounter "
+                                                     + "SET nextValue = nextValue + 1 "
+                                                     + "WHERE name = ${name} "
+                                                     + "      AND nextValue = ${value}")
+                                        .set("name", sequence)
+                                        .set("value", result.getNextValue())
+                                        .executeUpdate();
+                if (numRowsChanged == 1) {
+                    // Nobody else changed the counter, so we can savely return the determined value...
+                    return result.getNextValue();
+                }
+
+                // Block a short random amount of time to resolve conflicts with other waiting threads
+                Wait.randomMillis(50, 50);
```

Comment:
ordered chaos ... should this not at least have a smal span between min and max?